core/src/test/java/com/datastax/oss/driver/internal/core/ContactPointsTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.filter;
+import static org.junit.Assume.assumeTrue;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import com.datastax.oss.driver.api.core.metadata.EndPoint;
+import com.datastax.oss.driver.internal.core.metadata.DefaultEndPoint;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableSet;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.Set;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.slf4j.LoggerFactory;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ContactPointsTest {
+
+  @Mock private Appender<ILoggingEvent> appender;
+  @Captor private ArgumentCaptor<ILoggingEvent> loggingEventCaptor;
+  private Logger logger;
+  private Level initialLogLevel;
+
+  @Before
+  public void setup() {
+    logger = (Logger) LoggerFactory.getLogger(ContactPoints.class);
+    initialLogLevel = logger.getLevel();
+    logger.setLevel(Level.INFO);
+    logger.addAppender(appender);
+  }
+
+  @After
+  public void teardown() {
+    logger.detachAppender(appender);
+    logger.setLevel(initialLogLevel);
+  }
+
+  @Test
+  public void should_parse_ipv4_address_and_port_in_configuration() {
+    Set<EndPoint> endPoints =
+        ContactPoints.merge(Collections.emptySet(), ImmutableList.of("127.0.0.1:9042"), true);
+
+    assertThat(endPoints)
+        .containsExactly(new DefaultEndPoint(new InetSocketAddress("127.0.0.1", 9042)));
+  }
+
+  @Test
+  public void should_parse_ipv6_address_and_port_in_configuration() {
+    Set<EndPoint> endPoints =
+        ContactPoints.merge(Collections.emptySet(), ImmutableList.of("0:0:0:0:0:0:0:1:9042"), true);
+
```

Comment:
Shouldn't we add a couple more tests for ipv6? 
Maybe all tests could be parameterized with ipv4 and ipv6 to run variants for both?

Reply:
The only thing we need to cover is that we use the last colon to separate the port. The IPv6 parsing itself is delegated to the JDK, so we don't need to test different formats.

I've added the short format `::2` in the test, but that's the same really.